software/postgres/schema/gps.sql

```diff
@@ -0,0 +1,11 @@
+DROP TABLE IF EXISTS "gps";
```

Comment:
What are the "best practice" around database schema management? We will want to migrate a table structure without having to drop it one day -- with mysql I have always used a third party module to version and migrate databases, but that isn't likely to work well in our "micro-service" style design.

Reply:
So this is a file that you wrote that I have not . I assume it is used for the initial setup of the docker container?

For schema management, I have little experience with it and what it entails/requires. After a brief look, it appears we would be doing the same as mysql and using a 3rd party module like https://github.com/sqitchers/sqitch to do some of the work. Unsure how well it will work with our overall architecture

Reply:
Okay, I see. It is a great question on how we want to manage all this. The `postgres` init stuff only runs on the very first `up`, so that wont help with migrations ...

I guess we need services to be able to:

1. Create users for themselves (say a `gps` user which write access to `gps` table and a `oada` user with read access to `gps` and write access `oada` [or whatever that table to store "sent" is called])

2. Update / create tables on service startup (or somehow right before service startup)

I am working on docker-compose.yml files, container building, etc right now in #28, but I don't have any ideas for this. For now, we can just punt and get the db setup right with the postgres init, but any updates will need a solution.